chill-scala/src/main/scala/com/twitter/chill/KryoBase.scala

```diff
@@ -20,18 +20,29 @@ import com.esotericsoftware.kryo.KryoException
 import com.esotericsoftware.reflectasm.ConstructorAccess
 import com.esotericsoftware.kryo.serializers.FieldSerializer
 import _root_.java.lang.Thread
+
 import org.objenesis.instantiator.ObjectInstantiator
 import org.objenesis.strategy.InstantiatorStrategy
-
 import _root_.java.lang.reflect.{Constructor, Modifier}
 
+import com.esotericsoftware.kryo.ClassResolver
+import com.esotericsoftware.kryo.ReferenceResolver
+import com.esotericsoftware.kryo.util.DefaultClassResolver
+import com.esotericsoftware.kryo.util.MapReferenceResolver
+
 import scala.util.{Failure, Success, Try}
 
 /*
  * This is the base class of Kryo we use to fix specific scala
  * related issues discovered (ideally, this should be fixed in Kryo)
  */
-class KryoBase extends Kryo {
+class KryoBase(classResolver: ClassResolver, referenceResolver: ReferenceResolver)
+    extends Kryo(classResolver, referenceResolver) {
+  // no-args constructor instead of constructor with default args for binary compatibility
+  def this() {
```

Comment:
```suggestion
  def this() = {
```

I think we should avoid procedure syntax. Also 🎨 we can remove the outer `{}`

Reply:
is this procedure syntax? It doesn't return Unit, it is making a new instance of the class.

Reply:
yeah, I can't get an alternate syntax to work... for an example at the REPL.

Reply:
yes right! it's not really a procedure, but i think that kinda of syntax should be avoid whenever possible. regarding your error that's interesting I tried in 2.11 and 2.12 and had no issues.

```scala
[info] Starting scala interpreter...
Welcome to Scala 2.11.12 (OpenJDK 64-Bit Server VM, Java 1.8.0_222).
Type in expressions for evaluation. Or try :help.

scala> import com.twitter.chill._
import com.twitter.chill._

scala> import com.esotericsoftware.kryo.util.DefaultClassResolver
import com.esotericsoftware.kryo.util.DefaultClassResolver

scala> import com.esotericsoftware.kryo.util.MapReferenceResolver
import com.esotericsoftware.kryo.util.MapReferenceResolver

scala> new KryoBase()
res0: com.twitter.chill.KryoBase = com.twitter.chill.KryoBase@23c54aa6

scala> new KryoBase(new DefaultClassResolver, new MapReferenceResolver)
res1: com.twitter.chill.KryoBase = com.twitter.chill.KryoBase@68c54aa4
```

```scala
[info] Starting scala interpreter...
Welcome to Scala 2.12.8 (OpenJDK 64-Bit Server VM, Java 1.8.0_222).
Type in expressions for evaluation. Or try :help.

scala> import com.twitter.chill._
import com.twitter.chill._

scala> import com.esotericsoftware.kryo.util.DefaultClassResolver
import com.esotericsoftware.kryo.util.DefaultClassResolver

scala> import com.esotericsoftware.kryo.util.MapReferenceResolver
import com.esotericsoftware.kryo.util.MapReferenceResolver

scala> new KryoBase(new DefaultClassResolver, new MapReferenceResolver)
res0: com.twitter.chill.KryoBase = com.twitter.chill.KryoBase@58a12ac4

scala> new KryoBase()
res1: com.twitter.chill.KryoBase = com.twitter.chill.KryoBase@5b6e7715
```

Reply:
Sorry, I didn't mean this code didn't work, I mean `def this() = ` didn't work.

I agree in scala every class should have 1 constructor, but binary compatibility is very important, so this is worth breaking the rule.

Reply:
🤔 but those REPL where ran with `def this() =` I'm a bit puzzled now

Reply:
Huh. Well if it works great. I prefer your syntax. I must have messed something up.

Reply:
Glad to hear you solved it, and that it works fine!

Reply:
@CMoH we still need the CLA signed, just a nagging reminder (also, I think @regadas would like you to update the code to use the `=`).